airbyte-api-server/src/main/openapi/api.yaml

```diff
@@ -0,0 +1,723 @@
+      type: object
+      properties:
+        basicSchedule:
+          required:
+            - timeUnit
+            - units
+          type: object
+          properties:
+            timeUnit:
+              enum:
+                - minutes
+                - hours
+                - days
+                - weeks
+                - months
+              type: string
+            units:
+              format: int64
+              type: integer
+        cron:
+          required:
+            - cronExpression
+            - cronTimeZone
+          type: object
+          properties:
+            cronExpression:
+              type: string
+            cronTimeZone:
+              type: string
+    ConnectionScheduleType:
+      description: determine how the schedule data should be interpreted
+      enum:
+        - manual
+        - basic
+        - cron
+      type: string
+    NamespaceDefinitionType:
+      description: Method used for computing final namespace in destination
+      default: source
+      enum:
+        - source
+        - destination
+        - customformat
+      type: string
+    ConnectionStatus:
+      description: Active means that data is flowing through the connection. Inactive means it is not. Deprecated means the connection is off and cannot be re-activated. the schema field describes the elements of the schema that will be synced.
+      enum:
+        - active
+        - inactive
+        - deprecated
+      type: string
+    ConnectionUpdate:
+      type: object
+      allOf:
+        - required:
+            - sourceId
+            - destinationId
+            - status
+          type: object
+          properties:
+            name:
+              description: Optional name of the connection
+              type: string
+            namespaceDefinition:
+              $ref: "#/components/schemas/NamespaceDefinitionType"
+            namespaceFormat:
+              description: 'Used when namespaceDefinition is ''customformat''. If blank then behaves like namespaceDefinition = ''destination''. If "${SOURCE_NAMESPACE}" then behaves like namespaceDefinition = ''source''.'
+              type: string
+              example: "${SOURCE_NAMESPACE}"
+            prefix:
+              description: Prefix that will be prepended to the name of each stream when it is written to the destination.
+              type: string
+            sourceId:
+              format: uuid
+              type: string
+            destinationId:
+              format: uuid
+              type: string
+            schedule:
+              $ref: "#/components/schemas/ConnectionSchedule"
+            scheduleType:
+              $ref: "#/components/schemas/ConnectionScheduleType"
+            scheduleData:
+              $ref: "#/components/schemas/ConnectionScheduleData"
+            status:
+              $ref: "#/components/schemas/ConnectionStatus"
+        - $ref: "#/components/schemas/Resource"
+    ConnectionCreate:
```

Comment:
Do we need `ConnectionCreate` for our initial release. I think the whole schema / catalog is current missing in this object and is a required field for creating a connection. It is also a brutal object to model ([example](https://github.com/airbytehq/airbyte/blob/master/airbyte-api/src/main/openapi/config.yaml#L3304) to how it works in config api). If we _do_ need to do `ConnectionCreate`, we probably need to workshop that object.